docs/advanced/analytics/plugins.mdx

```diff
@@ -0,0 +1,340 @@
+  via plugins.
+---
+
+import AnalyticsFigure from "./assets/analytics.svg";
+import Figure from "@site/src/components/Figure";
+import ContactLink from "@site/src/components/ContactLink";
+import { useState } from "react";
+
+> _Since version 2.19_
+
+<p>{frontMatter.description}</p>
+
+Plugins are a powerful abstraction that let you:
+
+- add a new analytics provider (like google analytics or pixel)
+- hook into an existing analytics provider plugin
+- or add any kind of logic to react to visitor actions
+
+Plugins can be broken down into 2 types:
+
+- **Provider plugins** - connecting to third party analytic services
+- **Custom plugins** - additional features, data manipulation, & any other side
+  effects.
+
+Both have the same signature, as illustrated below.
+
+## Provider Plugins
+
+Let's say for this example we want to create a new provider plugin for a third
+party `acme` analytics tool.
+
+Provider plugins typically
+
+1. Load in the third party analytics script via initialize
+1. Use `track` or `page` events to send data into a third party analytics tool
+   (There is also another event `identify` which is not currently implemented in
+   Front-commerce)
+1. Have a loaded function to let analytics know when its safe to send the third
+   party data.
+
+Here is an example if the `analytics-plugin-acme` plugin:
+
+```js title="my-analytics-module/plugins/analytics-plugin-acme.js"
+// `settings` is the value defined in the plugin config in `config/analytics.js`
+export default function providerPluginExample(settings) {
+  // return object for analytics to use
+  return {
+    /* All plugins require a name */
+    name: "acme",
+    /* Everything else below this is optional depending on your plugin requirements */
+    config: {
+      whatEver: settings.whatEver,
+      elseYouNeed: settings.elseYouNeed,
+    },
+    initialize: ({ config }) => {
+      // load provider script to page
+    },
+    page: ({ payload }) => {
+      // call provider specific page tracking
+    },
+    track: ({ payload }) => {
+      // call provider specific event tracking
+    },
+    identify: ({ payload }) => {
+      // call provider specific user identify method
+    },
+    loaded: () => {
+      // return boolean so analytics knows when it can send data to third party
+      return !!window.myPluginLoaded;
+    },
+  };
+}
+```
+
+### Extending with `EcommercePlugin`
+
+The `EcommercePlugin` plugin will allow you to create an interface to map
+[e-commerce events and properties](https://developers.google.com/analytics/devguides/collection/ga4/ecommerce)
+to the relevant events and properties for your `acme` provider, you can find a
+full list of events in the
+[EcommercePlugin.js](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/src/web/core/analytics/plugins/e-commerce/domain/e-commerce-plugin.js)
+
+Below is an example of how to extend the `analytics-plugin-acme` plugin to
+handle e-commerce events.
```

Comment:
**issue:** I've struggled to understand in how it relates to `analytics-plugin-acme` from reading the example.\
Couldn't it be splitted into 2 different sections:
1. write the plugin
2. use this plugin as an enhancer of an existing one

(could also work in reverse order BTW)